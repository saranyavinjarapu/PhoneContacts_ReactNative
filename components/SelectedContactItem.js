import React from "react";
import { StyleSheet, View, Text } from "react-native";
import Colors from "../constants/Colors";

const SelectedContactItem = (props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.contactSelected}>
        {props.phoneContactSelected &&
          props.phoneContactSelected.map((contact, index) =>
            contact.phoneNumbers ? (
              <Text key={index} style={styles.contactNumber}>
                {contact.phoneNumbers[index].number}
              </Text>
            ) : (
              <Text key={index} style={styles.contactNumber}>
                No Contact Found
              </Text>
            )
          )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
  },
  contactSelected: {
    padding: 90,
    paddingLeft: 200,
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    width: "80%",
    maxWidth: "80%",
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 6,
    shadowOpacity: 0.26,
    backgroundColor: "white",
    elevation: 15,
    marginVertical: 100,
    borderRadius: 10,
  },
  contactNumber: {
    height: 50,
    width: 300,
    alignContent: "center",
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 10,
    fontSize: 30,
    color: Colors.primary,
  },
});

export default SelectedContactItem;
