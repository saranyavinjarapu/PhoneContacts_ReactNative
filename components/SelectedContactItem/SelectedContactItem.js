import React from "react";
import { View, Text } from "react-native";
import styles from "./SelectedContactItemStyles";

const SelectedContactItem = (props) => {
  return (
    <View style={styles.screen}>
      <View style={styles.contactSelected}>
        {props.phoneContactSelected &&
          props.phoneContactSelected.map((contact, index) =>
            contact.phoneNumbers ? (
              <Text key={index} style={styles.contactNumber}>
                {contact.phoneNumbers[index].number}
              </Text>
            ) : (
              <Text key={index} style={styles.contactNumber}>
                No Contact Found
              </Text>
            )
          )}
      </View>
    </View>
  );
};

export default SelectedContactItem;
