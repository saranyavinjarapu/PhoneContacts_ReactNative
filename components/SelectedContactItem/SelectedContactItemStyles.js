import { StyleSheet } from "react-native";
import Colors from "../../constants/Colors";

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    padding: 50,
    paddingTop: 1,
  },
  contactSelected: {
    flexDirection: "row",
    alignItems: "center",
    textAlign: "center",
    width: 300,
    justifyContent: "center",
    height: 200,
    shadowColor: "black",
    shadowOffset: { width: 0, height: 2 },
    shadowRadius: 20,
    shadowOpacity: 0.26,
    backgroundColor: "white",
    elevation: 15,
    borderRadius: 50,
    marginTop: 70,
  },
  contactNumber: {
    fontSize: 30,
    color: Colors.primary,
  },
});

export default styles;
