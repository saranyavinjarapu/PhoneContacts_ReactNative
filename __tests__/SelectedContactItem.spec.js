import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import SelectedContactItem from "../components/SelectedContactItem";

describe("Selected Contact Item Renders Correctly", () => {
  it("renders correctly", () => {
    const snap = render(<SelectedContactItem />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
