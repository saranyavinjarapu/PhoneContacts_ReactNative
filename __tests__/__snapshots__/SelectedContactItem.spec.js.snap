// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected Contact Item Renders Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 50,
      "paddingTop": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 50,
        "elevation": 15,
        "flexDirection": "row",
        "height": 200,
        "justifyContent": "center",
        "marginTop": 70,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.26,
        "shadowRadius": 20,
        "textAlign": "center",
        "width": 300,
      }
    }
  />
</View>
`;
